runelite-client/src/main/java/net/runelite/client/callback/Hooks.java

```diff
@@ -117,7 +124,31 @@ public static void draw(MainBufferProvider mainBufferProvider, Graphics graphics
 			log.warn("Error during overlay rendering", ex);
 		}
 
-		renderer.provideScreenshot(image);
+		// Stretch the game image if the user has that enabled
+		if (!client.isResized() && client.isStretchedEnabled())
+		{
+			Dimension stretchedDimensions = client.getStretchedDimensions();
+
+			BufferedImage newImage = new BufferedImage(stretchedDimensions.width, stretchedDimensions.height, BufferedImage.TYPE_INT_RGB);
+
+			Graphics2D g = (Graphics2D) newImage.getGraphics();
+			g.setRenderingHint(RenderingHints.KEY_INTERPOLATION,
+					client.isStretchedFast()
+						? RenderingHints.VALUE_INTERPOLATION_NEAREST_NEIGHBOR
+						: RenderingHints.VALUE_INTERPOLATION_BILINEAR);
+			g.drawImage(image, 0, 0, stretchedDimensions.width, stretchedDimensions.height, null);
+			g.dispose();
+
+			graphics.drawImage(newImage, 0, 0, client.getCanvas());
```

Comment:
these 2 parts (before else and after else) can be merged and just draw different image.

Reply:
Hardly, since "image" has to be preserved

Reply:
So this will not work? (after making the image non-final)

```diff
-			graphics.drawImage(newImage, 0, 0, client.getCanvas());
-			renderer.provideScreenshot(newImage);
+			image = newImage;
		}
-		else
-		{
-			graphics.drawImage(image, x, y, client.getCanvas());
-			renderer.provideScreenshot(image);
+		graphics.drawImage(image, x, y, client.getCanvas());
+		renderer.provideScreenshot(image);
```

Reply:
Well, with setting the x and y to 0 as well there.